tests/runtime/dwarf

```diff
@@ -128,6 +142,73 @@ REQUIRES_FEATURE dwarf
 TIMEOUT 5
 AFTER ./testprogs/inline_function
 
+NAME print uprobe arg as deref pointer - struct
+PROG uprobe:./testprogs/uprobe_test:uprobeFunction2 {
+       print(*args.foo1);
+       exit();
+     }
+EXPECT { .a = 123, .b = hello, .c = [1,2,3], .d = 1311768467463790320 }
+REQUIRES_FEATURE dwarf
+TIMEOUT 5
+AFTER ./testprogs/uprobe_test
+
+NAME uprobe arg as reference - int
+PROG uprobe:./testprogs/uprobe_test_cxx:cpp:uprobeFunction1 {
+       printf("x = %d\n", args.x);
+       exit();
+     }
+EXPECT x = 42
+REQUIRES_FEATURE dwarf
+TIMEOUT 5
+AFTER ./testprogs/uprobe_test_cxx
+
+NAME print uprobe arg as reference - struct
+PROG uprobe:./testprogs/uprobe_test_cxx:cpp:uprobeFunction1 {
+       print(args.foo);
+       exit();
+     }
+EXPECT_REGEX ^\{ \.a = 1, \.b = 2, \.c = 3, \.x = 0x[0-9a-f]{1,16} \}$
+REQUIRES_FEATURE dwarf
+TIMEOUT 5
+AFTER ./testprogs/uprobe_test_cxx
+
+NAME uprobe arg as reference - struct member
+PROG uprobe:./testprogs/uprobe_test_cxx:cpp:uprobeFunction1 {
+       printf("foo.a = %d\nfoo.b = %d\nfoo.c = %d\nfoo.x = %d\n",
+         args.foo.a, args.foo.b, args.foo.c, args.foo.x);
+       exit();
+     }
+EXPECT foo.a = 1
+       foo.b = 2
+       foo.c = 3
+       foo.x = 42
+REQUIRES_FEATURE dwarf
+TIMEOUT 5
+AFTER ./testprogs/uprobe_test_cxx
+
+NAME uprobe arg as reference - array
+PROG uprobe:./testprogs/uprobe_test_cxx:cpp:uprobeArray {
+       printf("arr[0] = %d, arr[1] = %d, arr[9] = %d\n",
+         args.array[0], args.array[1], args.array[9]);
+       exit();
+     }
+EXPECT arr[0] = 1, arr[1] = 2, arr[9] = 10
+REQUIRES_FEATURE dwarf
+TIMEOUT 5
+AFTER ./testprogs/uprobe_test_cxx
+
+NAME AST Semantic Analyser − convert reference into pointer
+RUN {{BPFTRACE}} --dry-run -d ast -e 'uprobe:./testprogs/uprobe_test_cxx:cpp:uprobeFunction1 { args.foo }'
+EXPECT AST after: Semantic
+       -------------------
+       Program
+        uprobe:./testprogs/uprobe_test_cxx:cpp:uprobeFunction1
+         dereference
+          . :: [struct Foo *, AS(user)]
+           builtin: args :: [struct uprobe:./testprogs/uprobe_test_cxx:cpp:uprobeFunction1_args, ctx: 1, AS(user)]
+           foo
+TIMEOUT 5
```

Comment:
The issue is the test relies on debug information, so it needs to run against `tests/testprogs/uprobe_test_cxx`. I don't see how I could have a `reference` in the AST otherwise. Could I pass an AST as an input, instead of a BpfTrace script?
Otherwise, I can update the unit tests, so they can access the `testprogs`, and have it run against `uprobe_test_cxx`.

Reply:
Parsing the string representation of an AST would be very nice, but we don't have that yet.

The other option is to mock the data types you need (see `tests/mocks.(h|cpp)`). I think we'll need to insert some dummy data into `bpftrace.structs`.

Reply:
This is exactly what the `test_dwarf` class from `tests/dwarf_common.h` is for. You'll probably need to introduce something like `semantic_analyser_dwarf` test class but I think that this should be doable. See e.g. `field_analyser_dwarf` test class in `tests/field_analyser.cpp` for inspiration.